### PR TITLE
Dependency update for patch release from NavNative and MapboxMaps

### PR DESCRIPTION
**Pull request by ShanMa1991**
### Description
This Pr is to cheery pick the dependency update PRs #4310 and #4315 to adopt the patch release of `NavNative v123.1.0` and `MapboxMaps v10.10.1 ` to the `main` branch.

### CODE REVIEW

File: MapboxCoreNavigation.podspec [hunk 1]
```diff
@@ -44,7 +44,7 @@ Pod::Spec.new do |s|
   s.requires_arc = true
   s.module_name = "MapboxCoreNavigation"
 
-  s.dependency "MapboxNavigationNative", "~> 123.0"
+  s.dependency "MapboxNavigationNative", "~> 123.1.0"
```
Comment (MaximAlien) [inline MapboxCoreNavigation.podspec:47]: I think it'd be better to use:
```suggestion
  s.dependency "MapboxNavigationNative", "~> 123.1"
```

File: MapboxNavigation.podspec [hunk 2]
```diff
@@ -44,7 +44,7 @@ Pod::Spec.new do |s|
   s.module_name = "MapboxNavigation"
 
   s.dependency "MapboxCoreNavigation", "#{s.version.to_s}"
-  s.dependency "MapboxMaps", "~> 10.10"
+  s.dependency "MapboxMaps", "~> 10.10.1"
```
Comment (MaximAlien) [inline MapboxNavigation.podspec:47]: Same here:
```suggestion
  s.dependency "MapboxMaps", "~> 10.10"
```

Comment (ShanMa1991) [reply MapboxNavigation.podspec:47]: If we use `~> 10.10` in the `podspec`, the CocoaPod dependency will be `10.10` for `CocoaPodsTest`

File: MapboxCoreNavigation.podspec (hunk 1 above)
Comment (ShanMa1991) [reply MapboxCoreNavigation.podspec:47]: Thanks for the suggestion, I'll update that.

File: MapboxNavigation.podspec (hunk 2 above)
Comment (MaximAlien) [reply MapboxNavigation.podspec:47]: That's strange, I've just tried with default podspec where `s.dependency "MapboxMaps", "~> 10.10"` is used and the most recent version was resolved in Podfile.lock (10.10.1).

Comment (ShanMa1991) [reply MapboxNavigation.podspec]: Thanks for the suggestion, issue was solved.

**Review by MaximAlien — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**